Collage refactor

AndreasMadsen opened the pull request:
* Refactors collage to scale the logo more dynamically. The previous version had some heavy jitter when the resolution was changed.
* Most importantly, downsize all profile pictures dramatically. The resolution is still 2x the size and therefore renders beautifully on high-resolution displays (phones etc). The size and compression before were just much higher than what could be displayed. 
* Use lazy-loading for all profile pictures.
* Use `.webp` format, with `.jpg` as a fallback, for improved compression.

This should dramatically reduce the loading time of the font page and member page, which previously could be several seconds. -- The current download size is 7.4MB, this reduces it to 187kb.

dalafh commented on _data/people.yaml:
```diff
@@ -211,7 +211,7 @@
 - name: Shabnam Haghzare
   title: intern
-  picture: Shabnam_Haghzare.JPG
+  picture: shabnam_haghzare
```
Is there a reason why Shabnam is the only one with an _ in her name?

AndreasMadsen replied:
I don't think so. The file was named this way before I took over.

dalafh replied:
Ohhh okay yeah

dalafh commented on _includes/collage.html:
```diff
@@ -1,18 +1,20 @@
-                style="grid-column-start: {{person.collage_column_start}};"
-                {% endif %}
-            ><img src="/assets/images/bios/{{person.picture}}"></div>
+            {% if person.collage_column_start != nil %}
```
Is it possible to fix the alignment from line 8 to 10 since it's in the div tag?

AndreasMadsen replied:
Fixed.

dalafh commented on _includes/collage.html:
```diff
@@ -1,18 +1,20 @@
+            style="grid-column-start: {{person.collage_column_start}};"
+            {% endif %}
+            >
+            <picture>
```
Same for lines 12 to 16 that are between the two div tags

AndreasMadsen replied:
Fixed.

dalafh commented on _includes/collage.html:
```diff
@@ -1,18 +1,20 @@
+            >
+            <picture>
+                <source srcset="/assets/images/bios_compressed/{{person.picture}}.webp" type="image/webp">
+                <source srcset="/assets/images/bios_compressed/{{person.picture}}.jpg" type="image/jpeg">
```
Are the lines 14 and 15 redundant since they both use exactly the same source?

dalafh commented on _includes/person.html:
```diff
@@ -1,5 +1,9 @@
 {% if include.picture != "" and include.picture != nil %}
-    <img src="/assets/images/bios/{{include.picture}}">
+<picture>
```
Fixing alignment inside the if (lines 2 to 6)

AndreasMadsen replied:
Indentation of template files is never perfect. I general, the indentation should be done such the produced content is indented correctly.

dalafh replied:
Yeah okay that makes sense. I know it doesn't change the logic but Sarath said that the reviews needed to focus more on the coding practices than the logic so that's why I pointed it out. I agree that it's not necessary though

dalafh commented on _includes/person.html:
```diff
@@ -1,5 +1,9 @@
-    <img src="/assets/images/bios/{{include.picture}}">
+<picture>
+    <source srcset="/assets/images/bios_compressed/{{include.picture}}.webp" type="image/webp">
+    <source srcset="/assets/images/bios_compressed/{{include.picture}}.jpg" type="image/jpeg">
```
Same as before: are the lines 4 and 5 redundant since they both use exactly the same source?

AndreasMadsen replied:
No. The `src` in `img` is a fallback, in case the browser doesn't support `picture`.

dalafh commented on assets/css/style.scss:
```diff
@@ -66,7 +69,7 @@ ul.people li p {
 	margin-bottom: 20px;
 
 	display: grid;
-	grid-template-columns: auto auto;
```
Why is that necessary?

AndreasMadsen replied:
Because the two child nodes are now both flexible, hence auto will have arbitrary meaning. Before the size of the `title` child contained this enough to have a deterministic layout.

dalafh replied:
I see!

dalafh commented on assets/css/style.scss:
```diff
@@ -97,48 +100,12 @@ ul.people li p {
 
 @media only screen and (max-width : 570px) {
 	.collage {
-		margin-top: 0px;
```
Why is that necessary?

AndreasMadsen replied:
Fixed.

dalafh reviewed (commented):
Why are there white.jpg and white.webp in the assets/images/bios_compressed folder?

AndreasMadsen commented:
> Why are there white.jpg and white.webp in the assets/images/bios_compressed folder?

Previously the layout was dependent on everybody having a profile image. If someone didn't want that, they could use `white.png`. I've fixed it now such it is completely redundant and removed the files.

dalafh commented on assets/css/style.scss:
```diff
@@ -97,48 +100,13 @@ ul.people li p {
 @media only screen and (max-width : 570px) {
 	.collage {
 		margin-top: 0px;
-		margin-bottom: 20px;
```
Same for that one, sorry for noticing a little late...

AndreasMadsen replied:
This line didn't do anything, because `margin-bottom: 20px` always applied.

dalafh replied:
Okay great then!

dalafh reviewed (approved):
All good!
